include/aos/common/tools/time.hpp

```diff
@@ -18,14 +18,245 @@
 namespace aos {
 
 /**
- * Base type for a time duration in nanoseconds. Can also be negative to set a point back in time.
+ * Size of a time in string representation.
  */
-using Duration = int64_t;
+const auto cTimeStrLen = AOS_CONFIG_TIME_STR_LEN;
 
 /**
- * Size of a time in string representation.
+ * Base type for a time duration in nanoseconds. Can also be negative to set a point back in time.
  */
-const auto cTimeStrLen = AOS_CONFIG_TIME_STR_LEN;
+class Duration {
+public:
+    /**
+     * Default constructor.
+     */
+    constexpr Duration() = default;
+
+    // cppcheck-suppress noExplicitConstructor
+    /**
+     * Constructs object instance.
+     *
+     * @param duration duration in nanoseconds.
+     */
+    constexpr Duration(int64_t duration)
+        : mDuration(duration)
+    {
+    }
+
+    /**
+     * Returns duration in nanoseconds.
+     *
+     * @result int64_t.
+     */
+    int64_t Nanoseconds() const { return mDuration; }
+
+    /**
+     * Returns duration in microseconds.
+     *
+     * @result int64_t.
+     */
+    int64_t Microseconds() const { return Nanoseconds() / 1000; }
+
+    /**
+     * Returns duration in milliseconds.
+     *
+     * @result int64_t.
+     */
+    int64_t Milliseconds() const { return Microseconds() / 1000; }
+
+    /**
+     * Returns duration in seconds.
+     *
+     * @result float.
+     */
+    float Seconds() const { return Milliseconds() / 1000; }
```

Comment:
Should be `Nanoseconds() / 1000000000` also add .0 or f: `Nanoseconds() / 1000000000.0`. Otherwise, it perform int division and then cast to float if I'm not mistaken.